pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -831,3 +831,14 @@ def test_mi_add_cell_missing_row_non_unique():
         columns=MultiIndex.from_product([[1, 2], ["A", "B"]]),
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_loc_get_scalar_casting_to_float():
+    # GH#41369
+    df = DataFrame(
+        {"a": 1.0, "b": 2}, index=MultiIndex.from_arrays([[3], [4]], names=["c", "d"])
+    )
+    result = df.loc[(3, 4), "b"]
+    assert result == 2
```

Comment:
this test passes on master. 2.0 == 2 is True.

Reply:
Added an isinstance check